bpf/bpf_host.c

```diff
@@ -1034,6 +1034,17 @@ int to_netdev(struct __ctx_buff *ctx __maybe_unused)
 	 (defined(ENABLE_DSR) && defined(ENABLE_DSR_HYBRID)) || \
 	 defined(ENABLE_MASQUERADE) || \
 	 defined(ENABLE_EGRESS_GATEWAY))
+#ifdef ENABLE_EGRESS_REVNAT_AT_HOST
+	/* Do conntrack lookup on all egress traffic, if its nodeport
+	 * traffic do reverse NAT.
+	 */
+	ret = egress_rev_nodeport(ctx);
+	if (IS_ERR(ret)) {
+		return send_drop_notify_error(ctx, 0, ret,
+						CTX_ACT_DROP,
+						METRIC_EGRESS);
+	}
+#endif /* ENABLE_EGRESS_REVNAT_AT_HOST */
 	if ((ctx->mark & MARK_MAGIC_SNAT_DONE) != MARK_MAGIC_SNAT_DONE) {
 		ret = handle_nat_fwd(ctx);
```

Comment:
Wouldn't it better to integrate egress_rev_nodeport() into handle_nat_fwd()? What is the blocker for merging the two?

Reply:
For couple of reason, it's not merged/put together.
1) By keeping it separate, it improves the code readability. Separate handler for Nodeport traffic vs non-Nodeport traffic.

2) NodePort NAT translation based upon Conntrack tables, handle_nat_fwd translations based upon  few map tables. 

I'm not sure what would be the gain of putting together of both functions.